### ISSUE TITLE
[BUG] Postgrey rejects mails although disabled

### ISSUE THREAD

**Issue by ontheair81** [docker-mailserver/docker-mailserver#2671]
### Miscellaneous first checks

- [X] I checked that all ports are open and not blocked by my ISP / hosting provider.
- [X] I know that SSL errors are likely the result of a wrong setup on the user side and not caused by DMS itself. I'm confident my setup is correct.

### Affected Component(s)

Postgrey is disabled, but still rejects mails

### What happened and when does this occur?

```Markdown
Although `ENABLE_POSTGREY=0` is set, mails from one specific sender is not accepted but rejected.
```

### What did you expect to happen?

```Markdown
I expect postgrey not rejecting mails when disabled.
```

### How do we replicate the issue?

```Markdown
For me this issue just affects one specific sender, this is the Debian Security mailing list.

1. Subscribe to the [mailing list](https://lists.debian.org/debian-security-announce/).
2. Reply to the confirmation mail to confirm the subscription.
3. Wait for the first message sent by the mailing list. It will not hit your inbox but be rejected instead.
```

### DMS version

v11.1.0

### What operating system is DMS running on?

Linux

### What instruction set architecture is DMS running on?

x86_64 / AMD64

### What container orchestration tool are you using?

Docker Compose

### docker-compose.yml

```yml
version: "3.7"

services:

    mailserver:
        image: "docker.io/mailserver/docker-mailserver:11.1.0"
        container_name: "mailserver"
        # If the FQDN for your mail-server is only two labels (eg: example.com),
        # you can assign this entirely to `hostname` and remove `domainname`.
        hostname: "mail"
        domainname: "[MYDOMAIN]"
        env_file: "/home/user/docker_compose/mailserver/mailserver.env"
        # More information about the mail-server ports:
        # https://docker-mailserver.github.io/docker-mailserver/edge/config/security/understanding-the-ports/
        # To avoid conflicts with yaml base-60 float, DO NOT remove the quotation marks.
        ports:
           - "25:25"     # SMTP  (explicit TLS => STARTTLS)
           - "143:143"   # IMAP4 (explicit TLS => STARTTLS)
           - "465:465"   # ESMTP (implicit TLS)
           - "587:587"   # ESMTP (explicit TLS => STARTTLS)
           - "993:993"   # IMAP4 (implicit TLS)
           - "4190:4190" # ManageSieve
        volumes:
           - "/home/user/bind-mounts/dms/mail-data/:/var/mail/"
           - "/home/user/bind-mounts/dms/mail-state/:/var/mail-state/"
           - "/home/user/bind-mounts/dms/mail-logs/:/var/log/mail/"
           - "/home/user/bind-mounts/dms/config/:/tmp/docker-mailserver/"
           - "/home/user/docker_compose/mailserver/policyd-spf.conf:/etc/postfix-policyd-spf-python/policyd-spf.conf"
           - "/home/user/bind-mounts/certstore:/etc/letsencrypt/live/:ro"
        restart: "always"
        stop_grace_period: "1m"
        cap_add:
          - "NET_ADMIN"
          - "SYS_PTRACE"
        dns:
          - "9.9.9.9"
          - "8.8.8.8"
          - "1.1.1.1"
```

### Relevant log output

```bash
Jul  2 08:26:48 mail postfix/smtp[899663]: 45649C1BC6: to=<debian-security-announce-request@lists.debian.org>, relay=bendel.debian.org[82.195.75.100]:25, delay=5.5, delays=0.01/0/2.2/3.3, dsn=4.2.0, status=deferred (host bendel.debian.org[82.195.75.100] said: 450 4.2.0 <debian-security-announce-request@lists.debian.org>: Recipient address rejected: Greylisted, see http://postgrey.schweikert.ch/help/lists.debian.org.html (in reply to RCPT TO command))
```

### Other relevant information

```Markdown
This is the relevant part from the env file:

`ENABLE_POSTGREY=0
# greylist for N seconds
POSTGREY_DELAY=300
# delete entries older than N days since the last time that they have been seen
POSTGREY_MAX_AGE=35
# response when a mail is greylisted
POSTGREY_TEXT="Delayed by Postgrey"
# whitelist host after N successful deliveries (N=0 to disable whitelisting)
POSTGREY_AUTO_WHITELIST_CLIENTS=5`
```

### What level of experience do you have with Docker and mail servers?

- [ ] I am inexperienced with docker
- [ ] I am inexperienced with mail servers
- [ ] I am uncomfortable with the CLI

### Code of conduct

- [X] I have read this project's [Code of Conduct](https://github.com/docker-mailserver/docker-mailserver/blob/master/CODE_OF_CONDUCT.md) and I agree
- [X] I have read the [README](https://github.com/docker-mailserver/docker-mailserver/blob/master/README.md) and the [documentation](https://docker-mailserver.github.io/docker-mailserver/edge/) and I searched the [issue tracker](https://github.com/docker-mailserver/docker-mailserver/issues?q=is%3Aissue) but could not find a solution

**Comment by casperklein** [comment]
From your log I would assume, the grey listing happens on the remote server (bendel.debian.org).

**Comment by ontheair81** [comment]
Oh, I see. Thank you for the clarification!
While the first messages produced this kind of logs, now it seems to work and I am getting messages from this mailinglist.
